File: CustomerService/src/main/java/com/mwaf/customerservice/service/CustomerService.java
package com.mwaf.customerservice.service;




import java.util.List; import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import com.mwaf.customerservice.model.Customer;
import com.mwaf.customerservice.repository.CustomerRepository;

@Service public class CustomerService {

    @Autowired
    private CustomerRepository customerRepository;

    public Customer createCustomer(Customer customer) {
        return customerRepository.save(customer);
    }

    public Optional<Customer> getCustomerById(Long id) {
        return customerRepository.findById(id);
    }

    public List<Customer> getAllCustomers() {
        return customerRepository.findAll();
    }

    public Customer updateCustomer(Long id, Customer customerDetails) {
        Customer customer = customerRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Customer not found with id " + id));
        customer.setName(customerDetails.getName());
        customer.setEmail(customerDetails.getEmail());
        customer.setPhone(customerDetails.getPhone());
        customer.setAddress(customerDetails.getAddress());
        return customerRepository.save(customer);
    }

    public void deleteCustomer(Long id) {
        customerRepository.deleteById(id);
    }

    public Optional<Customer> findByUserId(Long userId) {
        return customerRepository.findByUserId(userId);
    }
}

File: SmartFrontend-main/src/services/customerService.js
// src/services/customerService.js
import { customerApi } from './api';


export function getAllCustomers() {
    return customerApi.get('');
}

export function getCustomerById(customerId) {
    return customerApi.get(`/${customerId}`);
}

export function updateCustomer(customerId, customerData) {
    return customerApi.put(`/${customerId}`, customerData);
}

export function deleteCustomer(customerId) {
    return customerApi.delete(`/${customerId}`);
}

export function getCustomerByUserId(userId) {
    return customerApi.get(`/byUserId/${userId}`);
}




//
// export function getAllCustomers() {
//     return customerApi.get('/api/customer');
// }
//
// export function getCustomerById(customerId) {
//     return customerApi.get(`/api/customers/${customerId}`);
// }
//
// export function updateCustomer(customerId, customerData) {
//     return customerApi.put(`/api/customers/${customerId}`, customerData);
// }
//
// export function deleteCustomer(customerId) {
//     return customerApi.delete(`/api/customers/${customerId}`);
// }
//
// export function getCustomerByUserId(userId) {
//     return customerApi.get(`/api/customers/byUserId/${userId}`);
// }